inventory_management_system_api/repositories/system.py

```diff
@@ -171,12 +172,14 @@ def _is_duplicate_system(self, parent_id: Optional[str], code: str) -> bool:
 
     def _has_child_elements(self, system_id: CustomObjectId) -> bool:
         """
-        Check if a System has any child System's based on its ID
+        Check if a System has any child System's or any Item's based on its ID
 
         :param system_id: ID of the System to check
         :return: True if the System has child elements, False otherwise
         """
         logger.info("Checking if system with ID '%s' has child elements", str(system_id))
-        # Check if it has System's
-        system = self._systems_collection.find_one({"parent_id": system_id})
-        return system is not None
+
+        return (
+            self._systems_collection.find_one({"parent_id": system_id}) is not None
+            or self._items_collection.find_one({"system_id": system_id}) is not None
+        )
```

Comment:
I like this as it won't execute the second query if the first returns. We should do the same for catalogue category.

Reply:
Made a comment about this in https://github.com/ral-facilities/inventory-management-system-api/pull/148#discussion_r1450366862